pandas_log/aop_utils.py

```diff
@@ -94,28 +100,40 @@ def _get_param_value(param_with_default, arg_value):
     return f"{_get_bold_text(fn.__name__)}({args_vals}):"
 
 
-def restore_pandas_func_copy(func, prefix=settings.ORIGINAL_METHOD_PREFIX):
+def _raise_on_bad_class(cls):
+    implemented_classes = (pd.DataFrame, pd.Series)
+    if cls not in implemented_classes:
+        raise TypeError("cls must be one of {}".format(implemented_classes))
+
+
+def restore_pandas_func_copy(
+    cls, func, prefix=settings.ORIGINAL_METHOD_PREFIX
+):
     """ Restore the original pandas method instead of overridden one
 
+        :param cls: class containing the method
         :param func: pandas method name
         :param prefix: the prefix used to keep original method
         :return: None
     """
 
-    original_method = getattr(pd.DataFrame, func)
-    setattr(pd.DataFrame, func.replace(prefix, ""), original_method)
+    _raise_on_bad_class(cls)
+    original_method = getattr(cls, func)
+    setattr(cls, func.replace(prefix, ""), original_method)
 
 
-def keep_pandas_func_copy(func, prefix=settings.ORIGINAL_METHOD_PREFIX):
+def keep_pandas_func_copy(cls, func, prefix=settings.ORIGINAL_METHOD_PREFIX):
     """ Saved copy of the pandas method before it overridden
 
+        :param cls: class containing the method
         :param func: pandas method name
         :param prefix: the prefix used to keep original method
         :return: None
     """
 
-    original_method = getattr(pd.DataFrame, func)
-    setattr(pd.DataFrame, f"{prefix}{func}", original_method)
+    _raise_on_bad_class(cls)
```

Comment:
@cmdavis4 when will this happen ?

Reply:
It won't within the other code of this package, but since it's a function without a preceding underscore I treated it as something an end-user would potentially use (however unlikely that may be) and just wanted to make it a bit more type safe.